### PR TITLE
Update systemd service

### PR DESCRIPTION
**Pull request by ewilly**
## Problem

- *Description of why you made this PR*

## Solution

- *And how do you fix that problem*

## PR Status

- [ ] Code finished and ready to be reviewed/tested
- [ ] The fix/enhancement were manually tested (if applicable)

## Automatic tests

Automatic tests can be triggered on https://ci-apps-dev.yunohost.org/ *after creating the PR*, by commenting "!testme", "!gogogadgetoci" or "By the power of systemd, I invoke The Great App CI to test this Pull Request!". (N.B. : for this to work you need to be a member of the Yunohost-Apps organization)

### CODE REVIEW

**Comment by ewilly** [comment]
!testme

**Comment by ewilly** [comment]
!testme

**Comment by ewilly** [comment]
!testme

**Comment by ewilly** [comment]
!testme

File: conf/systemd.service [hunk 1]
```diff
@@ -1,16 +1,14 @@
...
 User=__APP__
-WorkingDirectory=__PATH__
-ExecStart=/opt/yunohost/__APP__/bin/hass --config "__PATH__" --verbose
-StandardOutput=append:/var/log/__APP__.log
-StandardError=inherit
+WorkingDirectory=__DATAPATH__/.__APP__
+ExecStart=__FINALPATH__/bin/hass --config "__DATAPATH__/.__APP__" --log-file "__LOGFILE__" --verbose
 
-# Sandboxing options to harden security
+# Sandboxing options to harden security__DATA_PATH__
```
Comment (alexAubin) [inline conf/systemd.service:11]: Uh, typo ?

File: conf/systemd.service [hunk 2]
```diff
@@ -1,16 +1,14 @@
...
-After=network-online.target
+After=network.target
 
 [Service]
 Type=simple
 User=__APP__
-WorkingDirectory=__PATH__
-ExecStart=/opt/yunohost/__APP__/bin/hass --config "__PATH__" --verbose
-StandardOutput=append:/var/log/__APP__.log
-StandardError=inherit
```
Comment (alexAubin) [inline conf/systemd.service:11]: Why remove these though ?

File: scripts/install [hunk 3]
```diff
@@ -61,31 +68,38 @@ ynh_exec_fully_quiet myynh_install_homeassistant --path="$data_path"
 # set default configuration files and move all homeassistant_conf_files
 ynh_script_progression --message="Configuring the installation..."
 ynh_replace_string --match_string="__PORT__" --replace_string="$port" --target_file="../conf/homeassistant_conf_files/configuration.yaml"
-ynh_replace_string --match_string="__PATH__" --replace_string="$data_path/.$app" --target_file="../conf/homeassistant_conf_files/configuration.yaml"
+ynh_replace_string --match_string="__APP__" --replace_string="$app" --target_file="../conf/homeassistant_conf_files/configuration.yaml"
+ynh_replace_string --match_string="__DATAPATH__" --replace_string="$data_path" --target_file="../conf/homeassistant_conf_files/configuration.yaml"
 cp -r "../conf/homeassistant_conf_files/." "$data_path/.$app/"
```
Comment (alexAubin) [inline scripts/install:73]: Sounds like you could really simplify this with `ynh_add_config`

File: scripts/upgrade [hunk 4]
```diff
@@ -28,24 +42,35 @@ if [ -f "/etc/yunohost/apps/$app/scripts/backup" ] ; then
...
 # stop systemd service
 ynh_script_progression --message="Stoping service..."
-ynh_systemd_action --service_name="$app@$app" --action=stop --line_match="Stopped Home Assistant" --log_path="/var/log/$app.log" --timeout=300
+ynh_systemd_action --service_name="$app" --action=stop --line_match="Stopped Home Assistant" --log_path="$log_file" --timeout=300
+
+# migrate to new app architecture
+ynh_script_progression --message="If needed, migrating to new app architecture..."
+if [ ! -d "$data_path" ]; then
+  # move $data_path to new directory
+  mv "/""home""/$app" "$data_path"
```
Comment (alexAubin) [inline scripts/upgrade:57]:
```suggestion
  mv "/home/$app" "$data_path"
```

**Review by alexAubin — APPROVED** [review]
Quick review, LGTM

File: conf/systemd.service (hunk 1 above)
Comment (ewilly) [reply conf/systemd.service:11]: yes corrected thanks

File: conf/systemd.service (hunk 2 above)
Comment (ewilly) [reply conf/systemd.service:11]: to avoid duplicate log entries: the app already redirect service log

File: scripts/install (hunk 3 above)
Comment (ewilly) [reply scripts/install:73]: seems that `ynh_add_config` only wor kwith file but i need to move a whole directory

File: scripts/upgrade (hunk 4 above)
Comment (ewilly) [reply scripts/upgrade:57]: to avoid linter detection that give a warning about /home but which is not relevant in these case

**Comment by ewilly** [comment]
!testme

File: scripts/install (hunk 3 above)
Comment (tituspijean) [reply scripts/install:73]: Alex was referring to the multiple `ynh_replace_string` commands. ;)

Comment (ewilly) [reply scripts/install:73]: hum i not sur to understand your advice, is it that ?
`cp -r "../conf/homeassistant_conf_files/." "$data_path/.$app/"`
`ynh_add_config --template="$data_path/.$app/configuration.yaml" --destination="$data_path/.$app/configuration.yaml"`

**Comment by ewilly** [comment]
rrrr ci can't execute the check because of `ERROR: Could not install packages due to an EnvironmentError: [Errno 122] Disk quota exceeded` (see [https://ci-apps-dev.yunohost.org/ci/logs/homeassistant_amd64_stable_complete.log](https://ci-apps-dev.yunohost.org/ci/logs/homeassistant_amd64_stable_complete.log))

File: scripts/install (hunk 3 above)
Comment (alexAubin) [reply scripts/install:73]: Yes, or using `./conf/homeassistant_conf_files/configuration.yaml` as `--template` in the second command (same result, but it's better from a semantic point of view)

(As a side note I'm super confused that the conf happens to be in `$data_path/.$app` x_x, I'm not sure if this is a deliberate choice or just the app internal structure imposing this)

Comment (ewilly) [reply scripts/install:73]: ok but it seems i need to keep 2 ynh_replace_string :
`cp -r "../conf/homeassistant_conf_files/." "$data_path/.$app/"`
`ynh_replace_string --match_string="__PORT__" --replace_string="$port" --target_file="../conf/homeassistant_conf_files/configuration.yaml"`
`ynh_replace_string --match_string="__DATAPATH__" --replace_string="$data_path" --target_file="../conf/homeassistant_conf_files/configuration.yaml"`
`ynh_add_config --template="../conf/homeassistant_conf_files/configuration.yaml" --destination="$data_path/.$app/configuration.yaml"`
is that fine ?
Regarding `$data_path/.$app` it's the default app structure (never try too move it but can try)

Comment (alexAubin) [reply scripts/install:73]: Why would you need to keep those 2 replace_string ? `ynh_add_config` will take care of replacing `__PORT__` with `$port` ... and should be replaced `__DATAPATH__` in the template by `__DATA_PATH__` (note the middle `_`) to be replaced by `$data_path`

**Pull request merged**